### PR TITLE
Bug 13233 : Upload size limit not verified

### PR DESCRIPTION
**Pull request by SilverDav**
Also, clarifications have been made on the functioning of the "Downloadable original image" instance parameter.

### CODE REVIEW

File: gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag [hunk 1]
```diff
@@ -171,6 +172,7 @@
       </c:if>
 
       var dateErrors;
+      let errorPromise = sp.promise.resolveDirectlyWith();
```
Comment (mmoqui) [inline gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag:175]: Be cautious: this var is overriden by your code if the if-statement is validated.
In order to keep the idea behind, the above code should be replaced by this one:
```javascript
let errorPromises = [sp.promise.resolveDirectlyWith()];
```
So, your promise can be added into the array. See below my next comment

File: gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag [hunk 2]
```diff
@@ -180,6 +182,7 @@
         errorMsg += "<li>" + error.message + "</li>";
         errorNb++;
       });
+      errorPromise = verifyFileInput($media);
```
Comment (mmoqui) [inline gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag:185]: With the modification above, this code can be now:
```javascript
errorPromises.push(verifyFileInpyt($media));
```
See below my next comment to know how to use this array of promises.

File: gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag [hunk 3]
```diff
@@ -193,22 +196,23 @@
...
-              "<b><fmt:message key="GML.ThisFormContains"/> 1 <fmt:message key="GML.error"/> : </b><ul>" +
-              errorMsg + "</ul>";
-          jQuery.popup.error(errorMsg);
-          break;
-        default :
-          errorMsg = "<b><fmt:message key="GML.ThisFormContains"/> " + errorNb +
-              " <fmt:message key="GML.errors"/> :</b><ul>" + errorMsg + "</ul>";
-          jQuery.popup.error(errorMsg);
-      }
+      errorPromise.then(function () {
```
Comment (mmoqui) [inline gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag:199]: This code can be now:
```javascript
sp.promise.whenAllResolved(errorPromises).then(function() {
...
});
```

**Review by mmoqui — CHANGES_REQUESTED** [review]

File: gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag (hunk 1 above)
Comment (mmoqui) [reply gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag:175]: After seeing with Yohann who has proposed you the solution, the better way is to initialize an empty array:
```javascript
let errorPromises = [];
```

File: gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag (hunk 3 above)
Comment (SilverDav) [reply gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag]: Done.

File: gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag (hunk 1 above)
Comment (SilverDav) [reply gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag]: Done.

File: gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag (hunk 3 above)
Comment (SilverDav) [reply gallery/gallery-war/src/main/webapp/WEB-INF/tags/silverpeas/gallery/editMedia.tag]: Done.

**Review by mmoqui — APPROVED** [review]

**Comment by mmoqui** [comment]
PR modified and merged by hand

**Pull request closed without merging**